### PR TITLE
Feature/add mtl example

### PR DESCRIPTION
**Pull request by sebastian-hatala-leanix**
With this PR I propose to add an example for Multi Task Learning (see https://github.com/pyg-team/pytorch_geometric/discussions/3966)

### CODE REVIEW

**Comment by wsad1** [comment]
@sebastian-hatala-leanix Is this very different from the example [here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/ppi.py) other than having some more exclusive parameters between tasks?. Please let me know if I am missing something.

**Comment by sebastian-hatala-leanix** [comment]
@wsad1 as you mentioned these two are very similar - in fact I copied the `ppi.py` and only added the body-head architecture for Multi Task Learning.

The result is also not much different. But after all this is meant to be a blueprint to those who want to tackle a problem with multiple tasks.

To give an example when this architecture is actually useful: We had a setting with three tasks of which one was a binary classification and two multi-class classifications. There having exclusive parameters and being able to use a different loss function per task led to the best result (as opposed to this problem where there is only binary tasks) 💪

**Comment by rusty1s** [comment]
@sebastian-hatala-leanix Thanks for the example. Is there any chance we can enhance this example with a dataset for which one needs to use different loss functions? While I like to include an example for multi-task learning, the current solution is obviously less efficient than just computing outputs at once, leading to showcase „bad-practices“.

**Comment by sebastian-hatala-leanix** [comment]
Jup fair point, I'll have another look at the datasets and give it another go 👍

**Comment by rusty1s** [comment]
I‘ll be happy to discuss more. Feel free to leave this PR open.

**Comment by sebastian-hatala-leanix** [comment]
Thanks! So I can think of two areas of problems where MTL with GCN is really efficient & effective:
1. When predicting a set of independent targets given a common input (e. g. generating metadata from pictures like [here](https://towardsdatascience.com/tuning-a-multi-task-fate-grand-order-trained-pytorch-network-152cfda2e086))
2. When trying to generate embeddings so that nodes are semantically comparable.

I am not sure yet, if any of the `PPI`, `Yelp`, `QM7b`, `QM9` datasets - which are the ones with more than one task - are really suited for this. If not I'll look beyond the data in PyG - maybe someone has a recommendation?

I'll reopen this once we found a good match, how does that sound?

**Comment by sebastian-hatala-leanix** [comment]
**Update** I think I found two approaches to this I'd like to pursue:

1. Given a dataset with a single classification task one could add one head that tries to reconstruct the (sub-)graph using an autoencoder. This could work on e.g. the `IMDb` dataset.
2. Another interesting problem for the approach could be Knowledge Graphs like `WordNet18RR` one head could learn to predict the relation and another one to predict the fact.

Sounds a bit like I have a hammer and am looking for a nail here. But I guess that is sort of what is happening here...😅

**Comment by rusty1s** [comment]
I think those are good options. Alternatively, one can have a look at related literature for applying Graph Neural Networks in a multi-task learning scenario, *e.g.*, [this one](https://arxiv.org/pdf/1809.07695.pdf) or [this one](https://grlearning.github.io/papers/132.pdf), where additional tasks could involve learning some graph statistics.
